File: Life/Infrastructure/DataObjects/Cell.cs
namespace Infrastructure.DataObjects {
    public class Cell {
        public virtual int X { get; set; }
        public virtual int Y { get; set; }

        protected bool Equals(Cell other) {
            return X == other.X && Y == other.Y;
        }

        public override bool Equals(object obj) {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((Cell) obj);
        }

        public override int GetHashCode() {
            unchecked {
                return (X*397) ^ Y;
            }
        }
    }
}

File: life-java/src/main/java/ru/agiledays/life/server/infrastructure/dataobjects/Cell.java
package ru.agiledays.life.server.infrastructure.dataobjects;

public class Cell {
    private int x;
    private int y;
    
    public int getX() {
        return x;
    }
    
    public void setX(final int x) {
        this.x = x;
    }
    
    public int getY() {
        return y;
    }
    
    public void setY(final int y) {
        this.y = y;
    }
    
    public boolean equals(Cell otherCell) {
        return otherCell != null && otherCell.x == x && otherCell.y == y;
    }
    
    @Override
    public boolean equals(Object other) {
        if (other != null && Cell.class.isAssignableFrom(other.getClass())) {
            return equals((Cell)other);
        }
        
        return false;
    }

    @Override
    public int hashCode() {
        return (x * 397) ^ y;
    }
    
    public Cell() {}
    
    public Cell(final int row, final int column) {
        x = column;
        y = row;
    }
}
